### PR TITLE
Nativefy remixes screen

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
Screen Shot 2022-09-09 at 2 25 12 PM

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx [hunk 1]
```diff
@@ -47,20 +53,38 @@ const useStyles = makeStyles(({ spacing, palette, typography }) => ({
...
+  const lineup = useSelector(getRemixesTracksLineup)
+  const count = useSelector(getCount)
+  const track = useSelector(getTrack)
+  const user = useSelector(getUser)
+  const dispatch = useDispatch()
 
   const styles = useStyles()
+  const { params } = useRoute<'TrackRemixes'>()
+
+  const trackId = params.id
```
Comment (nicoback2) [inline packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx:65]: The code I wrote assumes that `id` is always present + a valid track id - is that correct?

Comment (sliptype) [reply packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx:65]: I believe that's correct!

File: packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx [hunk 2]
```diff
@@ -47,20 +53,38 @@ const useStyles = makeStyles(({ spacing, palette, typography }) => ({
...
+    dispatch(fetchTrack({ id: trackId }))
+    dispatch(
+      tracksActions.fetchLineupMetadatas(0, 10, false, {
+        trackId: trackId ?? null
+      })
+    )
+  }, [dispatch, trackId])
+
+  useEffect(() => {
+    return function cleanup() {
```
Comment (sliptype) [inline packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx:76]: Could this be part of the previous useEffect?

**Review by sliptype — APPROVED** [review]
Awesome!

File: packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx:76]: this is cool!

File: packages/web/src/common/store/pages/remixes-page/sagas.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const { fetchTrack, fetchTrackSucceeded } = remixesPageActions
+
+function* watchFetch() {
+  yield takeEvery(
+    fetchTrack.type,
+    function* (action: ReturnType<typeof fetchTrack>) {
+      yield call(waitForBackendSetup)
+      const { handle, slug, id } = action.payload
+      if (!id && (!handle || !slug)) {
+        throw new Error(
```
Comment (dylanjeffers) [inline packages/web/src/common/store/pages/remixes-page/sagas.ts:21]: nice

**Review by dylanjeffers — APPROVED** [review]
really great cleanup along with the nativifying

File: packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx (hunk 2 above)
Comment (sliptype) [reply packages/mobile/src/screens/track-screen/TrackRemixesScreen.tsx:76]: Yeah! I might use this same approach for the profile screen lineup reset that needs to happen

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--remix-screen

**Pull request merged**